src/main/kotlin/org/acejump/session/Session.kt

```diff
@@ -231,13 +232,15 @@ class Session(private val editor: Editor) {
    * Should only be used from [SessionManager] to dispose a
    * successfully ended session.
    */
-  internal fun dispose() {
+  internal fun dispose(jumpPerformed: Boolean) {
+    val (mark, query) = calculateMarkAndQuery(jumpPerformed)
+
     tagger = Tagger(editor)
     EditorKeyListener.detach(editor)
     tagCanvas.unbind()
     textHighlighter.reset()
     EditorCache.invalidate()
-    listeners.forEach(AceJumpListener::finished)
+    listeners.forEach { it.finished(mark, query) }
```

Comment:
Maybe this can be moved somewhere around here, where the query text and tag are directly available: 

https://github.com/acejump/AceJump/blob/2c0a236ed84df23b5ab353657fe251157b567d9f/src/main/kotlin/org/acejump/search/Tagger.kt#L56